code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -222,12 +222,18 @@
 	item_state = "coffee"
 	spillable = TRUE
 
-/obj/item/reagent_containers/food/drinks/mug/on_reagent_change(changetype)
+/obj/item/reagent_containers/food/drinks/mug/tea/on_reagent_change(changetype)
 	if(reagents.total_volume)
 		icon_state = "tea"
 	else
 		icon_state = "tea_empty"
 
+/obj/item/reagent_containers/food/drinks/mug/coco/on_reagent_change(changetype)
+	if(reagents.total_volume)
+		icon_state = "coco"
```

Comment:
Add a `var/icon_state_filled = "tea"` line to the mug typepath, change line 227 to `icon_state = icon_state_filled`, add an `icon_state_filled = "coco"` line to the mug/coco typepath and finally delete this proc.

Reply:
I couldnt figure this out because I'm dumb so I had Nebula help me, ended up making it so all mugs will show their reagent colors now instead of it being static.